### ISSUE TITLE
Hidden Fails To hide file on Windows SP1 x64

### ISSUE THREAD

**Issue by GhostNaix** [JKornev/hidden#9]
**UAC:** Disabled
**TestSigning:** On
**Driver(Hidden.sys):** Loaded in a service Successfully 
Hidden seems to fail to hide specified files even after the system is restarted and the driver is loaded
After launching **HiddenTests.exe** the following results were displayed:

**HiddenTests.exe output**

```
C:\Hidden-master>HiddenTests.exe
--------------------------------
File-System monitor tests result:
--------------------------------
Test 1: create single file, hide it, unhide it
Error, Hid_AddHiddenFile() failed with code: 3221225851
 failed!
--------------------------------
Registry monitor tests result:
--------------------------------
Test 1: create single reg key, hide it, unhide it
 successful!
Test 2: create single reg value, hide it, unhide it
 successful!
--------------------------------
Process monitor prot tests result:
--------------------------------
Test 1: attach, test, detach protection
 successful!
Test 2: create process, protect, check, unprotect
 successful!
--------------------------------
Process monitor excl tests result:
--------------------------------
Test 1: hide file, add excluded process, check file
Error, Hid_AddHiddenFile() failed with code: 3221225851
 failed!

C:\Hidden-master>
```

**Screenshot:**

![capture](https://user-images.githubusercontent.com/20989997/42822667-f96d3508-8a1e-11e8-9691-27e780361eef.JPG)

If I'm doing something wrong please correct me
Thank you

**Comment by JKornev** [comment]
Hi, thanks for the report, I'll check it soon.

**Comment by GhostNaix** [comment]
I'd also like to report that hidden 32-bit is also producing the same result

```
C:\Users\Tester\Desktop\x86>HiddenTests.exe
--------------------------------
File-System monitor tests result:
--------------------------------
Test 1: create single file, hide it, unhide it
Error, Hid_AddHiddenFile() failed with code: 3221225851
 failed!
--------------------------------
Registry monitor tests result:
--------------------------------
Test 1: create single reg key, hide it, unhide it
 successful!
Test 2: create single reg value, hide it, unhide it
 successful!
--------------------------------
Process monitor prot tests result:
--------------------------------
Test 1: attach, test, detach protection
 successful!
Test 2: create process, protect, check, unprotect
 successful!
--------------------------------
Process monitor excl tests result:
--------------------------------
Test 1: hide file, add excluded process, check file
Error, Hid_AddHiddenFile() failed with code: 3221225851
 failed!

C:\Users\Tester\Desktop\x86>
```

**Comment by JKornev** [comment]
Sorry for a huge delay, I tried to replicate the issue but nothing happens to me, an error you reported seems strange. In any case I'll keep this issue opened. Please let me know if you still have a reproduction.

**Comment by GhostNaix** [comment]
@JKornev What is your test environment config

**Comment by vasiliyph** [comment]
Check DebugView and when load driver saw error:
![screenshot_2](https://user-images.githubusercontent.com/22670109/50444746-a1871280-0923-11e9-8d8a-323d272a4f3f.png)

**Comment by JKornev** [comment]
So I have tried to replicate it again and nothing happens, I tried different environments w7x64, w7x86, w10x64rs5, w10x86rs5 . I guess that the error happens here https://github.com/JKornev/hidden/blob/master/Hidden/ExcludeList.c#L107. Therefore if that is true an error might be a result of an issue during an initialization. The only way to investigate it is to attach a kernel debugger and gather a debug output while hidden.sys initialization and HiddenTest execution.

Therefore if you are able to do it please do the following:
1. Compile and install a driver with the latest changes (git pull)
2. Attach a kernel debugger
3. Enable debug log output by the following debugger command:
_ed ed_IHVDRIVER_mask f_
4. Start a driver (sc start hidden)
5. Run HiddenTests.exe
6. Collect output from debugger

**Comment by GhostNaix** [comment]
Do I run the kernel debugger in live mode ?
because the command `ed ed_IHVDRIVER_mask f`
seems to yield

```
0: kd> ed ed_IHVDRIVER_mask f
Couldn't resolve error at 'ed_IHVDRIVER_mask f'
```

**Comment by GhostNaix** [comment]
Upon loading the starting the service containing hidden.sys, my kernel debugger threw this:
`hidden!InitializeFSMiniFilter[irql:0,pid:4][error]: Error, can't register filter, code:c0000034`

**Comment by JKornev** [comment]
Sorry not **ed ed_IHVDRIVER_mask f** but **ed kd_IHVDRIVER_mask f**

**Comment by JKornev** [comment]
Yup seems like an error c0000034 is a case why test fails on Hid_AddHiddenFile()

**Comment by JKornev** [comment]
According to MSDN this error means the following:

> STATUS_OBJECT_NAME_NOT_FOUND The filter service key is not found in the registry.  -or-  The filter instance is not registered.

I guess that is because you installed a driver by a wrong way. A valid installation flow is the following:
1. Compile "Hidden Package"
2. Copy files from "Release\Hidden Package" to system where you want to install a driver
3. Click right mouse button on Hidden.inf and choose "Install"

Please confirm did you do the same or not?

**Comment by JKornev** [comment]
Yes you right I'll add an installation section to readme
